src/main/java/ch/njol/skript/entity/FallingBlockData.java

```diff
@@ -112,18 +111,22 @@ protected boolean match(final FallingBlock entity) {
 		return true;
 	}
 
-	@SuppressWarnings("deprecation")
 	@Override
 	@Nullable
 	public FallingBlock spawn(Location loc, @Nullable Consumer<FallingBlock> consumer) {
 		ItemType t = CollectionUtils.getRandom(types);
 		assert t != null;
-		ItemStack i = t.getRandom();
-		if (i == null || i.getType() == Material.AIR || !i.getType().isBlock()) {
-			assert false : i;
+		Material material = t.getMaterial();
+
+		if (material == Material.AIR || !material.isBlock()) {
+			assert false : t;
```

Comment:
Is there a parse-time check that disallows `falling air`, cause I only see something for checking it the material can be a block. If not, we could add it, or we should not assert false if the material is air

Reply:
The condition also checks for AIR, which is a block, which isn't checked for at parsetime

Reply:
Ah I see what you ment.

Reply:
Looking thru the CraftBukkit/Minecraft, there is nothing that checks for AIR (or the like), therefor no errors.
The only error is making sure the material is a block. 

That said, we COULD add more conditions, but since the server allows it, I don't see why we would need to check for it.